src/ui/windows/TogglDesktop/TogglDesktop/Toggl.cs

```diff
@@ -15,6 +15,54 @@ namespace TogglDesktop
 {
 public static partial class Toggl
 {
+    #region Data types
+
+    public class TimelineEventView
+    {
+        public string Title;
+        public string Filename;
+        public Int64 Duration;
+        public string DurationString;
+        public bool Header;
+        // references subevents
+        public List<TimelineEventView> SubEvents;
+
+        public TimelineEventView(TogglTimelineEventView eventView)
+        {
+            Title = eventView.Title;
+            Filename = eventView.Filename;
+            Duration = eventView.Duration;
+            DurationString = eventView.DurationString;
+            Header = eventView.Header;
+            SubEvents = marshalList<TimelineEventView, TogglTimelineEventView>(eventView.Event, n => n.Event, t => new TimelineEventView(t));

```

Comment:
instead of `n => n.Event` there should be `n => n.Next`. this fixes the issue with every event showing only one subevent

Reply:
```suggestion
            SubEvents = marshalList<TimelineEventView, TogglTimelineEventView>(eventView.Event, n => n.Next, t => new TimelineEventView(t));

```

Reply:
Thank you, this saved me a lot of time 🙇‍♀️